cpp/src/Slice/Parser.cpp

```diff
@@ -3318,45 +3318,53 @@ Slice::Operation::outParameters() const
 }
 
 ParameterList
-Slice::Operation::sortedReturnAndOutParameters(const string& returnsName)
+Slice::Operation::returnAndOutParameters(const string& returnsName)
 {
-    bool shouldEscapeReturnsName = false;
+    ParameterList outParams = outParameters();
 
-    // First sort each parameter into either 'required' or 'optional'.
-    ParameterList required, optional;
-    for (const auto& param : outParameters())
+    // Check if this operation has a non-void return type.
+    if (_returnType)
     {
-        if (param->optional())
-        {
-            optional.push_back(param);
-        }
-        else
+        // Then check if any of the out parameter's mapped names collide with the return name.
+        bool shouldEscapeReturnsName = false;
```

Comment:
I really want to remove all this escaping logic from the Slice compilers.

All the escaping should be performed by the user with `xxx:identifier`. Missed escaping => generated code doesn't compile.

Reply:
This is a little different, since it's used by _us_ to check whether _our_ parameter should be escaped, not a user-provided identifier.

I agree that user's shouldn't be using language keywords, and if they want to, they can use `xxx:identifier` for that.

But expecting them to remap a parameter because in the internal generated code, we happen to have a temporary variable named `result`... I find this less compelling. These names are implementation details that users shouldn't have to know or care about. To accomplish that, we need to escape these variables if they collide with user's identifiers.

Reply:
If it's an truly internal name, we should use an invalid name like `$return`. Then no possible collision.

Reply:
Yes, we probably should be doing this.
I'll look into removing these vestiges of "escaping" after finishing the `xxx:identifier`s.